app/models/token.rb

```diff
@@ -21,21 +21,25 @@ def payload
                  end
   rescue JWT::ExpiredSignature => e
     Rails.logger.info(e.message, token: @token_string)
-    raise error_klass("Validation error: #{e.message}")
+    raise error_klass(e.message)
+  rescue JWT::DecodeError => e
+    raise error_klass(e.message)
   end
 
   def public_key
     decoded_token = JWT.decode(@token_string, nil, false, algorithm: 'RS256')
     kid = decoded_token[1]['kid']
-    OIDC::KeyService.get_key(kid)
+    key = OIDC::KeyService.get_key(kid)
+    raise error_klass("Public key not found for kid specified in token: '#{kid}'") if key.blank?
```

Comment:
should we raise this error to the user or log it? I'm not sure how sensitive a kid is

Reply:
I updated this to do both. The kid isn't sensitive at all + it's actually coming from the token they send us. So I don't think there's any risk to including it in the error (but there's also not that much to gain, just might steer a future provider in the right direction while debugging).